### PR TITLE
Gateway fixes

### PR DESCRIPTION
**Pull request by shubham3121**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

- [x] Fix: Not able to login into Gateway as GUEST
- [x] Fix: Node peers not visible in case applied to gateway using the GuestClient
- [x] Fix: Double Wrapping of response in case of forwarded message
- [x] Fix: Message forwarding not working correctly for non `api_call` routes
     - APIs like login, register, metadata, not correctly forwarding API calls
     - [ ] Create a forward endpoint to handle all types of forwarding requests

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

**Comment by shubham3121** [comment]
Changes from PR https://github.com/OpenMined/PySyft/pull/7888 will be pulled into this one.

**Comment by iamtrask** [comment]
Hey @shubham3121 for my visibility could you expand upon this README so I know what's up here.

**Comment by shubham3121** [comment]
@iamtrask updated the readme

File: packages/syft/src/syft/client/client.py [hunk 1]
```diff
@@ -70,6 +73,34 @@ def upgrade_tls(url: GridURL, response: Response) -> GridURL:
     return url
 
 
+def forward_message_to_proxy(
```
Comment (rasswanth-s) [inline packages/syft/src/syft/client/client.py:76]: Good one, shubham, for the generic structure, the code looks really good.

**Review by rasswanth-s — APPROVED** [review]
Wonderful Work @shubham3121  🙌 
Kudos , for refactoring the forwarding endpoints in a nice generic way.

**Pull request merged**